src/map/battle.cpp

```diff
@@ -3259,7 +3265,7 @@ static void battle_calc_skill_base_damage(struct Damage* wd, struct block_list *
 					RE_ALLATK_ADDRATE(wd, sd->bonus.atk_rate);
 				}
 #ifndef RENEWAL
-				if(sd->bonus.crit_atk_rate && !skill_id && is_attack_critical(wd, src, target, skill_id, skill_lv, false)) { // add +crit damage bonuses here in pre-renewal mode [helvetica]
+				if(sd->bonus.crit_atk_rate && is_attack_critical(wd, src, target, skill_id, skill_lv, false)) { // add +crit damage bonuses here in pre-renewal mode [helvetica]
```

Comment:
This is missing now. You need to add a code like:
```
#ifndef RENEWAL
	// In Pre-Renewal skills cannot crit
	if( !skill_id ){
		return false;
	}
#endif
```

Reply:
Re-reading the code it might be covered by the NK check though, because the flag is not set in pre-re.

Reply:
i think it would make more sense for the flag to be set in pre-re skill_db.txt (after all, it IS critting in pre-re), and use #ifdef RENEWAL to make the crit atk rate bonus apply.

Reply:
checking the skills in ratemyserver / irowiki classic , those skills should crit
however i am not sure how accurate it is
but looking on secret's edit
https://github.com/rathena/rathena/commit/08d160aaac8cdd286bbafb32532abe42b8a42376
it look like the skills should have crit

Reply:
well , the check is covered by NK as @Lemongrass3110 mentioned , and for now seems no one reported that something wrong with the pre-re skills as they are , and this edit should not effect the pre-re anyway , so if someone reported about it , we can just change the pre-re/skill_db , but for now this would just fix the renewal issue